dandi/tests/test_upload.py

```diff
@@ -181,6 +181,12 @@ def test_upload_sync_folder(
         text_dandiset.dandiset.get_asset_by_path("subdir2/banana.txt")
 
 
+def test_upload_bids(mocker: MockerFixture, bids_dandiset: SampleDandiset) -> None:
+    iter_upload_spy = mocker.spy(LocalFileAsset, "iter_upload")
+    bids_dandiset.upload(existing="forced")
+    iter_upload_spy.assert_called()
```

Comment:
@jwodder what is the best way to extend this unittest to also query the server after upload to ensure that the specific assets were uploaded?  I have tried to add `local_dandi_api` fixture and use it (while in pdb) but failed so far.